codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java

```diff
@@ -213,7 +207,6 @@ public String structureShape(final StructureShape shape) {
               assertionRequired,
               writer,
               memberShape.isOptional(),
-              shapeName.contains(memberName) ||
```

Comment:
Is this safe? I think we did this because we found issue with AWS SDK pointer vs non pointer

Reply:
This was done to make codegen compatible with v1 sdk but was never fully tested. Now that we have moved on to v2 sdk this shouldn't be an issue and the CI is passing too.